lib/timers.js

```diff
@@ -497,8 +497,12 @@ exports.setInterval = function(callback, repeat, arg1, arg2, arg3) {
 };
 
 exports.clearInterval = function(timer) {
-  if (timer && timer._repeat) {
-    timer._repeat = null;
+  if (timer) {
+    // It's allowed to use clearInterval to kill a setTimeout, which has its
+    // _repeat attribute to `null` already.
+    if (timer._repeat) {
+      timer._repeat = null;
```

Comment:
In all honesty, I would almost just go with `exports.clearInterval = clearTimeout;` The distinction is pretty meaningless especially since they're now interchangeable. Unsetting `_repeat` doesn't do anything. Or we could do:

```js
exports.clearInterval = function clearInterval(timer) {
  clearTimeout(timer);
};
```

Mostly to appease stack traces, etc. with differing function names.